pywemo/discovery.py

```diff
@@ -91,3 +95,54 @@ def device_from_uuid_and_location(uuid, mac, location,
                           rediscovery_enabled=rediscovery_enabled)
 
     return None
+
+
+def hostname_lookup(hostname):
+    """Resolve a hostname into an IP address."""
+    try:
+        # The {host} must be resolved to an IP address; if this fails, this will
+        # throw a socket.gaierror.
+        host_address = gethostbyname(hostname)
+
+        # Reset {host} to the resolved address.
+        LOG.debug(
+            'Resolved hostname %s to IP address %s.',
+            hostname, host_address)
+        return host_address
+
+    except gaierror:
+        # The {host}-as-hostname did not resolve to an IP address.
+        LOG.debug(
+            'Could not resolve hostname %s to an IP address.',
+            hostname)
+        return None
+
+
+def setup_url_for_address(host, port):
+    """Determine setup.xml url for a given host and port pair."""
+
+    # Force hostnames into IP addresses
+    try:
+        # Attempt to register {host} as an IP address; if this fails ({host} is not an IP address),
+        # this will throw a ValueError.
+        ip_address(host)
+        is_hostname = False
```

Comment:
I don't think the `is_hostname` variable is needed - take another look at my suggestion.

Reply:
Makes sense.